File: backend/service/src/main/java/com/ftn/sbnz/model/user/Statistic.java
package com.ftn.sbnz.model.user;

import com.ftn.sbnz.model.plant.Plant;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Statistic {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long Id;

    @ManyToOne
    private Plant plant;
    @ManyToOne
    private User user;
    @OneToMany
    private List<PositiveReview> positive = new ArrayList<>();
    @OneToMany
    private List<NegativeReview> negative = new ArrayList<>();

    public void addPositive(PositiveReview positiveReview){
        this.positive.add(positiveReview);
    }
    public void addNegative(NegativeReview positiveReview){
        this.negative.add(positiveReview);
    }

    public Statistic(Plant plant, User user) {
        this.plant = plant;
        this.user = user;
    }
}

File: front/src/app/model/statistic.ts
import {Plant} from "./plant";
import {User} from "./user";
import {Symptom} from "./disease";

export class Statistic{
  id:number=0;
  plant:Plant = new Plant();
  user:User = new User();
  positive:PositiveReview[] = []
  negative:NegativeReview[] = []
}

export class PositiveReview{
  id:number=0;
  date:string = '';
  comment:string='';
}

export class NegativeReview{
  id:number=0;
  date:string = '';
  symptom:Symptom=new Symptom();
  handled:boolean = false;
  dateHandled:string = '';
}

